### PR TITLE
Add styles parameter to manual_legend

### PR DESCRIPTION
**Pull request by tktran**
This PR is intended to fully implement issue #566. It includes unit tests for the new feature, and a bug fix for the existing manual_legend test (which failed to enable drawing the legend in the test image). It does not include updated documentation, which I would gladly begin working on upon approval of this PR. Also open to your suggested changes.

Questions for the @DistrictDataLabs/team-oz-maintainers:
- I set the default color of a legend entry (when none is specified in the colors nor styles parameter) to black. Please advise if there is a better option.

Checklist:
- [x] _Is the commit message formatted correctly?_
- [ ] _Have you noted the new functionality/bugfix in the release notes of the next release?_
- [x] _Included a sample plot to visually illustrate your changes? (see baseline image)_
- [x] _Do all of your functions and methods have docstrings?_
- [x] _Have you added/updated unit tests where appropriate?_
- [x] _Have you updated the baseline images if necessary?_
- [x] _Have you run the unit tests using `pytest`?_
- [x] _Is your code style correct (are you using PEP8, pyflakes)?_
- [ ] _Have you documented your new feature/functionality in the docs?_

### CODE REVIEW

File: tests/test_gridsearch/test_base.py [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+    @pytest.mark.skipif(pd is None, reason="test requires pandas")
+    def test_pandas_integration(self):
+        """
+        Test GridSearchColorPlot on sklearn occupancy data set (as pandas df)
+        """
+        
+        X, y = load_occupancy(return_dataset=True).to_pandas()
+        X, y = X.head(1000), y.head(1000)
+
+        gs = self.gridsearchcv
```
Comment (rebeccabilbro) [inline tests/test_gridsearch/test_base.py:115]: Looks like this is one of the places that is causing the Travis errors; would you mind removing this unused import?

File: tests/test_gridsearch/test_base.py [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+
+    def test_numpy_integration(self):
+        """
+        Test GridSearchColorPlot on sklearn occupancy data set (as numpy df)
+        """
+
+        X, y = load_occupancy(return_dataset=True).to_numpy()
+        X, y = X[:1000], y[:1000]
+
+        gs = self.gridsearchcv
```
Comment (rebeccabilbro) [inline tests/test_gridsearch/test_base.py:129]: Looks like this is another one of the places that is causing the Travis errors; would you mind removing this unused import as well?

File: tests/test_gridsearch/test_base.py [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+"""
+
+# ##########################################################################
+# ## Imports
+# ##########################################################################
+
+import pytest
+
+from tests.base import VisualTestCase
+from tests.fixtures import Dataset, Split
```
Comment (rebeccabilbro) [inline tests/test_gridsearch/test_base.py:22]: Looks like the `Split` import is not used in the file; would you mind removing it to resolve the unused imports errors on Travis?

**Review by rebeccabilbro — COMMENTED** [review]
Hello again @tktran  &mdash;  first off, my sincere apologies for the slow response, it's been a busy few months! 

Secondly, thank you so much for your contribution! I took a preliminary look and your PR and the test failures on Travis, and it looks like they're caused by a few unused imports.

Are you still interested in working on this PR? I noticed that the branch that you were working in appears to have been deleted, but if you can recover it and make these small updates, we can proceed with reviewing your PR!

**Comment by tktran** [comment]
Please see this PR redone at #1117. This PR that you are looking at is a little messy, since you are reviewing both code that I committed as part of issue #556 (as intended) and issue #308 (not intended, should have been isolated to its corresponding PR #1097). The new PR leaves the latter code out. Thanks for your patience.

**Comment by tktran** [comment]
@rebeccabilbro And thank you for taking the time to review my code. You may be seeing a few more contributions from me in the coming months, as I attempt to build up a history of contributions to Numfocus projects in preparation for applications to GSoC 2021. (But I do already know that yellowbrick won't be participating).

**Pull request closed without merging**